spec/graphql/schema/printer_spec.rb

```diff
@@ -590,4 +590,20 @@
     context = { names: ["Varied", "Image", "Sub"] }
     assert_equal expected.chomp, schema.to_definition(context: context, except: except_filter)
   end
+
+  describe "#print_type" do
+    it "returns the type schema as a string" do
+      expected = <<SCHEMA
+# A blog post
+type Post {
+  id: ID!
+  title: String!
+  body: String!
+  comments: [Comment!]
+  comments_count: Int! @deprecated(reason: \"Use \\\"comments\\\".\")
+}
+SCHEMA
+      assert_equal expected.chomp, GraphQL::Schema::Printer.new(schema).print_type(schema.types['Post'])
```

Comment:
Since the public API is `BaseType#to_definition`, how about calling that instead? (In a sense, `Schema::Printer` is an implementation detail of that API.) This test could even go in `base_type_spec`.